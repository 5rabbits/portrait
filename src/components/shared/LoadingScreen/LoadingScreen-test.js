import { React, toJSON, mount } from 'test/helper'
import LoadingScreen from 'shared/LoadingScreen'

describe('LoadingScreen', () => {
  it('should render with minimum props', () => {
    const component = mount(
      <LoadingScreen />,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  describe('if props.text is defined', () => {
    it('should display the text container', () => {
      const component = mount(
        <LoadingScreen text="test" />,
      )

      expect(toJSON(component)).toMatchSnapshot()
    })
  })
})
