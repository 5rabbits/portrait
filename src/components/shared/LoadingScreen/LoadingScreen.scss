@import '~styles/core';

.LoadingScreen {
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.LoadingScreen__text {
  margin-top: 24px;
  color: $default-secondary-text-color;
}
