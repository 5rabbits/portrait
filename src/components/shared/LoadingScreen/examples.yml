-
  description: With default props.
  code: >
    <div style={{ height: 300 }}>
      <LoadingScreen />
    </div>

-
  description: With custom text.
  code: >
    <div style={{ height: 300 }}>
      <LoadingScreen text="Loading data..." />
    </div>

