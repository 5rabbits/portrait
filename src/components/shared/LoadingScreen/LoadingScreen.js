import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Container, Spinner } from 'shared'
import cx from 'classnames'
import './LoadingScreen.scss'

export default class LoadingScreen extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    text: PropTypes.string,
  }

  static defaultProps = {
    className: null,
    text: null,
  }

  render() {
    const { className, text } = this.props

    return (
      <Container
        className={cx('LoadingScreen', className)}
        >
        <Spinner
          size="5em"
          ringSize={4}
        />
        {text &&
          <div className="LoadingScreen__text">
            {text}
          </div>
        }
      </Container>
    )
  }
}
