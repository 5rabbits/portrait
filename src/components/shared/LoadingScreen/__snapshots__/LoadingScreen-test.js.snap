// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen if props.text is defined should display the text container 1`] = `
<LoadingScreen
  className={null}
  text="test"
>
  <div
    className="LoadingScreen"
  >
    <Spinner
      backgroundColor={null}
      className={null}
      color={null}
      ringSize={4}
      size="5em"
    >
      <div
        className="Spinner"
        style={
          Object {
            "borderColor": null,
            "borderLeftColor": null,
            "borderWidth": 4,
            "height": "5em",
            "width": "5em",
          }
        }
      />
    </Spinner>
    <div
      className="LoadingScreen__text"
    >
      test
    </div>
  </div>
</LoadingScreen>
`;

exports[`LoadingScreen should render with minimum props 1`] = `
<LoadingScreen
  className={null}
  text={null}
>
  <div
    className="LoadingScreen"
  >
    <Spinner
      backgroundColor={null}
      className={null}
      color={null}
      ringSize={4}
      size="5em"
    >
      <div
        className="Spinner"
        style={
          Object {
            "borderColor": null,
            "borderLeftColor": null,
            "borderWidth": 4,
            "height": "5em",
            "width": "5em",
          }
        }
      />
    </Spinner>
  </div>
</LoadingScreen>
`;
